layers/+completion/auto-completion/packages.el

```diff
@@ -109,11 +109,14 @@
 
       (add-hook 'company-completion-started-hook 'company-turn-off-fci)
       (add-hook 'company-completion-finished-hook 'company-maybe-turn-on-fci)
-      (add-hook 'company-completion-cancelled-hook 'company-maybe-turn-on-fci))
+      (add-hook 'company-completion-cancelled-hook 'company-maybe-turn-on-fci)
+
+      (add-hook 'text-mode-hook 'company-mode nil t))
     :config
     (progn
       (spacemacs|diminish company-mode " ⓐ" " a")
 
+      (spacemacs|add-company-backends :modes text-mode)
```

Comment:
I am not sure that this is a good idea. The general design of the auto completion layer is to not decide whether a layer registers company backends for its modes. When a company-backend is registered the macro `spacemacs|add-company-backends` will activate company mode for the registered modes automatically instead. Layers have to call `post-init-` code to register handlers on a layer basis. 

With this change we break this design by having the auto completion layer activate company for all `text-modes` in general. I think we should rather change the existing layers to properly register the backends for their respective modes this includes the respective `text-modes`.

Reply:
I went looking at this, but somehow I got distracted by other things. Anyway, thanks for the explanation about the general design, and the 'add-company-backend' macro. So, in any case, I will remove the hook then.

The thing I wanted to fix is auto-completion in text-mode buffers,
i.e. when I write some text in a .txt buffer, I would like to have autocompletion (and I think Spacemacs should offfer/activate it by default for this case).
As I could not find any specialized `text-layer`, I went looking for where best to add this. Maybe I should create a special layer for it, but I think that would be overkill. Also, I am not sure why we would not want basic auto-completion in all text layers. But if we for some reason don't want that, then it would still be nice to configure it for `.txt` buffers.
I find it just weird to have no auto-completion at all in such buffers.

Reply:
it seems that org-mode and tex-mode both derive from text-mode. so changing behavior of the text-mode will leak into other modes unexpectedly. (because Tex is a programming language 🤦🏻‍♂️ not natural language)

Reply:
I wanted to test if is this messes up something (because those backends get added to the layer anyway), then I found two thing about auto-complete:

[Here](https://develop.spacemacs.org/layers/+completion/auto-completion/README.html#enable-company-globally) the documentation says that enabling company globally is not recommended because disabling company in specific layers will not work anymore.

While [here]((https://develop.spacemacs.org/doc/DOCUMENTATION.html#disabling-layer-services-in-other-layers) the documentation mentions that company can be disabled for specific modes by loading the layer as follows:
`(auto-completion :disabled-for org)`.

But testing this on an `org` file, it seems that the disabling mechanism is already broken.

Finally, as far as I know about all this, in my opinion it would make sense to make auto-completion a default in text-mode (also when no 'human language editing' layer is installed), but some disabling mechanism should work. Also, there is still `fundamental-mode` if users really don't want auto-complete (by default).